test/endtoendtestutils.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) by Claudio Cambra <claudio.cambra@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "endtoendtestutils.h"
+
+#include <QNetworkProxy>
+#include <QJsonObject>
+#include <QSignalSpy>
+
+#include "cmd/simplesslerrorhandler.h"
+#include "creds/httpcredentials.h"
+#include "gui/accountmanager.h"
+#include "libsync/theme.h"
+#include "accessmanager.h"
+#include "httplogger.h"
+#include "syncenginetestutils.h"
+#include "testhelper.h"
+
+constexpr auto serverUrl = "https://server";
+
+/** End to end test credentials access manager class **/
+
+class EndToEndTestCredentialsAccessManager : public OCC::AccessManager
+{
+public:
+    EndToEndTestCredentialsAccessManager(const EndToEndTestCredentials *cred, QObject *parent = nullptr)
+        : OCC::AccessManager(parent)
+        , _cred(cred)
+    {
+    }
+
+protected:
+    QNetworkReply *createRequest(Operation op, const QNetworkRequest &request, QIODevice *outgoingData) override
+    {
+        QNetworkRequest req(request);
+        QByteArray credHash = QByteArray(_cred->user().toUtf8() + ":" + _cred->password().toUtf8()).toBase64();
+        req.setRawHeader("Authorization", "Basic " + credHash);
+
+        return OCC::AccessManager::createRequest(op, req, outgoingData);
+    }
+
+private:
+    // The credentials object dies along with the account, while the QNAM might
+    // outlive both.
+    QPointer<const EndToEndTestCredentials> _cred;
```

Comment:
not sure why you use a `QPointer` because you never test its validity
keep in mind that `QPointer` will be set to `nullptr` once the pointed object (if derived from a `QObject`) has been deleted

Reply:
just a pointer now

Reply:
Now we check for the validity